dask/utils.py

```diff
@@ -1755,6 +1755,9 @@ def parse_timedelta(s, default="seconds"):
 
     n = float(prefix)
 
+    if suffix.lower() not in timedelta_sizes:
+        valid_units = ", ".join(timedelta_sizes.keys())
+        raise ValueError(f"Invalid time unit: {suffix}. Valid units are: {valid_units}")
     multiplier = timedelta_sizes[suffix.lower()]
```

Comment:
I think it's better to keep this as a KeyError, and use try/catch rather than a pre-flight check

```suggestion
    try:
        multiplier = timedelta_sizes[suffix.lower()]
    except KeyError:
        valid_units = ", ".join(timedelta_sizes.keys())
        raise KeyError(f"Invalid time unit: {suffix}. Valid units are: {valid_units}") from None
```

Reply:
Sure, we can do that. I actually think `ValueError` is better (it's about the meaning of what the user did wrong, where `KeyError` feels like it's about the implementation of how we store units in a dict) but I care very little.